index.js

```diff
@@ -82,7 +84,7 @@ class PluginXrpPaychan extends PluginBtp {
       return [{
         protocolName: 'ripple_channel_id',
         contentType: BtpPacket.MIME_TEXT_PLAIN_UTF8,
-        data: Buffer.from(this._outgoingChannel)
+        data: Buffer.from(this._outgoingChannel || '')
```

Comment:
If this code is reached and `this._outgoingChannel` is null, then something is probably wrong

Reply:
What can happen is `_handleData` getting executed before `_connect` finished and, thus, `_outgoingChannel` is undefined.

Reply:
But if we set `_outgoingChannel` to `''`, then the peer will set their incoming channel to `''`

Reply:
As long as the peer does not overwrite its exist incoming channel, I don't think that's a problem. `''` is falsy, so the peer should just execute this [code](https://github.com/ripple/ilp-plugin-xrp-paychan/blob/3e77c594efa0bc990073f4db73a7032b9ced8984/index.js#L139-L142). Semantically, returning the empty string means "I don't have a paychan yet, try later again".

However, your comment reminds me that a peer MUST validate the incoming payment channel (see my comment on your [PR](https://github.com/ripple/ilp-plugin-xrp-paychan/pull/34))